### PR TITLE
refactor: remove CSS modules in global mode

### LINKED ISSUE #308: [Discussion] Deprecate :local
In my mind, it makes more sense to consider local-scope (and CSS Modules by extension) as something that's **always on or always off**.

I'm seeing a lot of folks hitting the problem where they enable `?modules` and then want to include non-local CSS (e.g. https://github.com/webpack/css-loader/issues/253) but never in the same file. I think @chrislloyd's suggestion in https://github.com/webpack/css-loader/pull/287 is a better approach, so you can configure two instances of `css-loader`, one local and one global, maybe having ".module.css" be local and ".css" be global (like https://github.com/gatsbyjs/gatsby/issues/193). 

In global CSS files, then, I don't see any reason to keep `:local`, except that that was how CSS Modules got started in the first place. But I'm interested in having a discussion about this because I want to know if anyone's still using `:local` and feels it should be kept.

### PR DESCRIPTION
**Pull request by geelen**
Fixes #308

### CODE REVIEW

File: lib/processCss.js [hunk 1]
```diff
@@ -147,34 +147,41 @@ module.exports = function processCss(inputSource, inputMap, options, callback) {
...
-			mode: options.mode,
-			rewriteUrl: function(global, url) {
-				if(parserOptions.url){
-					if(!loaderUtils.isUrlRequest(url, root)) {
-						return url;
-					}
-					if(global) {
-						return loaderUtils.urlToRequest(url, root);
+	var pipeline = postcss(
+		(options.mode === 'local' ? [
```
Comment (sokra) [inline lib/processCss.js]: I think there is also a `"pure"` mode which need the postcss plugins too.

**Comment by SpaceK33z** [comment]
@geelen, are you still interested in fixing this?

**Comment by michael-ciniawsky** [comment]
@geelen @sokra What's the status here? 😛

**Comment by michael-ciniawsky** [comment]
@geelen friendly ping, this one is actually really important && appreciated

**Comment by michael-ciniawsky** [comment]
Closed due to inactivity, eventually superseded by #497, feel free to reopen :)

**Pull request closed without merging**